pkg/kubectl/cmd/portforward.go

```diff
@@ -70,7 +76,7 @@ func NewCmdPortForward(f cmdutil.Factory, cmdOut, cmdErr io.Writer) *cobra.Comma
 		},
 	}
 	cmd := &cobra.Command{
-		Use: "port-forward POD [LOCAL_PORT:]REMOTE_PORT [...[LOCAL_PORT_N:]REMOTE_PORT_N]",
+		Use: "port-forward POD|RESOURCE/NAME [LOCAL_PORT:]REMOTE_PORT [...[LOCAL_PORT_N:]REMOTE_PORT_N]",
```

Comment:
if `kubectl port-forward <pod>` is now deprecated, we should remove it from the usage string.  It is also strange that we have a `-p` flag in addition to listing the resource in the arguments.  I'd suggest we hide/deprecate that flag also.  

Ideally we'd do some heuristicy stuff so that you could also do `kubectl port-forward service foo 8080` (notice no `/`).  That would require us to look to see if things look like a number to determine if it is the single-pod case or the type-space-name case.

Reply:
It is `-p` flag that is deprecated.  The message is

```
-p POD is DEPRECATED and will be removed in a future version. Use port-forward POD instead.
```

I am not familiar with the deprecation policy in this case.  I can remove the support for `-p` and add that to release note if that is the right thing to do for this PR.

Reply:
@jbeda how about modifying the message to say the `-p` flag will be removed in 1.11?